packages/sync-actions/test/product-sync-variants.spec.js

```diff
@@ -982,6 +982,33 @@ describe('Actions', () => {
     ])
   })
 
+  test('shouldnt thow `TypeError: Cannot read property 0 of undefined` if array props exist on before and doesnt exist on now object', async () => {
+    const before = {
+      categories: [],
+      masterVariant: {
+        id: 1,
+        key: 'default-masterVariant-key-de07e9bc-e352-422f-abe6-910d9879b995',
+        isMasterVariant: true,
+        prices: [],
+        images: [],
+        attributes: [],
+        assets: [],
+      },
+    }
+
+    const now = {
+      masterVariant: {
+        id: 1,
+        key: 'default-masterVariant-key-de07e9bc-e352-422f-abe6-910d9879b995',
+        isMasterVariant: true,
+      },
+    }
+
+    productsSync.buildActions(now, before)
+
+    expect(before).toMatchObject(now)
```

Comment:
Please have a look at the other tests in this file, we should do the assertion on the returning value from the `buildActions` method. Not on the variables we assigned within this test.

Reply:
Yep i did it like this beginning https://github.com/commercetools/nodejs/pull/1525/commits/625b1525e16fe9ba749351f5b66bea8d542128db

But the main point of the test is to compare the two objects to make sure that array props which exist on before will be coppied to now object to avoid the error `TypeError: Cannot read property 0 of undefined`

Reply:
I see. So we are mutating the original `now`?

Reply:
https://github.com/commercetools/nodejs/commit/625b1525e16fe9ba749351f5b66bea8d542128db looks like a good test to me 👍